### PR TITLE
[Staticroutes] allow http method configuration

### LINKED ISSUE #5938: [Staticroute] allow to define http method
Similar to how symfony routes do it.

Would you be interested in that feature? I would happy to create a PR.

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
Allow configurable HTTP Method for Staticroutes

Resolves #5938

### CODE REVIEW

File: models/Staticroute.php [hunk 1]
```diff
@@ -696,6 +701,33 @@ public function match($path, $params = [])
...
+    {
+        if ($this->methods && !is_array($this->methods)) {
+            $this->methods = explode(',', $this->methods);
+        }
+
+        return $this->methods;
+    }
+
+    /**
+     * @param string|array $methods
```
Comment (weisswurstkanone) [inline models/Staticroute.php]: return value missing

Comment (weisswurstkanone) [inline models/Staticroute.php]: return value missing

Comment (dpfaffenbauer) [reply models/Staticroute.php]: done

File: models/Staticroute.php [hunk 2]
```diff
@@ -696,6 +701,35 @@ public function match($path, $params = [])
...
+     *
+     * @return $this
+     */
+    public function setMethods($methods)
+    {
+        if (!is_array($methods)) {
+            $methods = strlen($methods) ? explode(',', $methods) : [];
```
Comment (brusch) [inline models/Staticroute.php]: Maybe we should `trim()` the values as well.

Comment (dpfaffenbauer) [reply models/Staticroute.php]: guess better to trim each item?

Comment (brusch) [reply models/Staticroute.php]: yep, that's actually what i meant 😊

Comment (dpfaffenbauer) [reply models/Staticroute.php]: done

**Comment by dpfaffenbauer** [comment]
@brusch docs are done as well

File: models/Staticroute.php [hunk 3]
```diff
@@ -77,6 +77,11 @@ class Staticroute extends AbstractModel
      */
     public $siteId;
 
+    /**
+     * @var array|string
```
Comment (weisswurstkanone) [inline models/Staticroute.php]: can this be a string?

Comment (dpfaffenbauer) [reply models/Staticroute.php]: yes, cause it is stored as a string, the getter converts it to an array

Comment (dpfaffenbauer) [reply models/Staticroute.php]: actually, not. you are right. I'll change it

Comment (dpfaffenbauer) [reply models/Staticroute.php]: changed it

File: lib/Routing/Staticroute/Router.php [hunk 4]
```diff
@@ -227,14 +227,23 @@ public function match($pathinfo)
      *
      * @return array
      */
-    protected function doMatch($pathinfo)
+    protected function doMatch($pathinfo, Request $request = null)
```
Comment (dvesh3) [inline lib/Routing/Staticroute/Router.php]: please add param in phpdoc

File: doc/Development_Documentation/02_MVC/04_Routing_and_URLs/02_Custom_Routes.md [hunk 5]
```diff
@@ -30,6 +30,7 @@ Following options are relevant:
 * *Defaults* - defaults for variables separated by | - e.g. key=value|key2=value2 
 * *Site* - Site for which this route should be applied to. 
 * *Priority* - priority in resolving the URL pattern. 
+* *Methods* - define which HTTP Methods are valid. You can define multiple ones at ones by using a comma as delimiter. If empty, all are allowed. (one of `HEAD`, `GET`, `POST`, `PUT`, `PATCH`, `DELETE`, `PURGE`, `OPTIONS`, `TRACE` or `CONNECT`)
```
Comment (dvesh3) [inline doc/Development_Documentation/02_MVC/04_Routing_and_URLs/02_Custom_Routes.md]: "ones at ones"

Comment (dpfaffenbauer) [reply doc/Development_Documentation/02_MVC/04_Routing_and_URLs/02_Custom_Routes.md]: done

File: lib/Routing/Staticroute/Router.php (hunk 4 above)
Comment (dpfaffenbauer) [reply lib/Routing/Staticroute/Router.php]: done

File: models/Staticroute.php [hunk 6]
```diff
@@ -696,6 +701,38 @@ public function match($path, $params = [])
...
+    /**
+     * @param array $methods
+     *
+     * @return $this
+     */
+    public function setMethods($methods)
+    {
+        if (!is_array($methods)) {
+            $methods = strlen($methods) ? explode(',', $methods) : [];
+            $methods = array_map(static function ($entry) {
```
Comment (brusch) [inline models/Staticroute.php]: Why not just `$methods = array_map('trim', $methods );` ?

Comment (dpfaffenbauer) [reply models/Staticroute.php]: because it really doesn't matter.

**Comment by dpfaffenbauer** [comment]
🎉 🍻

**Pull request merged**